nemo_text_processing/text_normalization/en/taggers/money.py

```diff
@@ -183,11 +182,16 @@ def __init__(self, cardinal: GraphFst, decimal: GraphFst, deterministic: bool =
         # weight for SH
         final_graph |= pynutil.add_weight(decimal_graph_with_minor, -0.0001)
 
+        # utilizes morphosyntactic features to append "per units"
+        graph_per_units = pynutil.insert(' morphosyntactic_features: "') + per_units + pynutil.insert('"')
+
         if not deterministic:
             final_graph |= integer_graph_reordered | decimal_default_reordered
             # to handle "$2.00" cases
             final_graph |= pynini.compose(
-                NEMO_SIGMA + pynutil.delete(".") + pynini.closure(pynutil.delete("0"), 1), integer_graph_reordered
+                NEMO_SIGMA + pynutil.delete(".") + pynini.closure(pynutil.delete("0"), 1), integer_graph_reordered,
             )
+
+        final_graph += graph_per_units.ques
```

Comment:
You want it recursive `per month per quarter`

Reply:
Good point. Let me work on that.